js_modules/dagster-ui/packages/ui-core/src/assets/AssetWipeDialog.tsx

```diff
@@ -1,75 +1,186 @@
+import {showCustomAlert} from '../app/CustomAlertProvider';
 import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
+import {PythonErrorInfo} from '../app/PythonErrorInfo';
 import {displayNameForAssetKey} from '../asset-graph/Utils';
+import {NavigationBlock} from '../runs/NavigationBlock';
+import {Inner, Row} from '../ui/VirtualizedTable';
+import {numberFormatter} from '../ui/formatters';
 
 interface AssetKey {
   path: string[];
 }
 
-export const AssetWipeDialog = ({
+const CHUNK_SIZE = 100;
+
+export const AssetWipeDialog = memo(
+  (props: {
+    assetKeys: AssetKey[];
+    isOpen: boolean;
+    onClose: () => void;
+    onComplete: (assetKeys: AssetKey[]) => void;
+    requery?: RefetchQueriesFunction;
+  }) => {
+    return (
+      <Dialog
+        isOpen={props.isOpen}
+        title="Wipe materializations"
+        onClose={props.onClose}
+        style={{width: '80vw', maxWidth: '1200px', minWidth: '600px'}}
+      >
+        <AssetWipeDialogInner {...props} />
+      </Dialog>
+    );
+  },
+);
+
+export const AssetWipeDialogInner = ({
   assetKeys,
-  isOpen,
   onClose,
   onComplete,
   requery,
 }: {
   assetKeys: AssetKey[];
-  isOpen: boolean;
   onClose: () => void;
   onComplete: (assetKeys: AssetKey[]) => void;
   requery?: RefetchQueriesFunction;
 }) => {
   const [requestWipe] = useMutation<AssetWipeMutation, AssetWipeMutationVariables>(
     ASSET_WIPE_MUTATION,
     {
-      variables: {assetKeys: assetKeys.map(asAssetKeyInput)},
       refetchQueries: requery,
     },
   );
 
+  const [isWiping, setIsWiping] = useState(false);
+  const [wiped, setWiped] = useState(0);
+
+  const didCancel = useRef(false);
   const wipe = async () => {
     if (!assetKeys.length) {
       return;
     }
-    await requestWipe();
+    setIsWiping(true);
+    for (let i = 0, l = assetKeys.length; i < l; i += CHUNK_SIZE) {
+      if (didCancel.current) {
+        return;
+      }
+      const nextChunk = assetKeys.slice(i, i + CHUNK_SIZE);
+      const result = await requestWipe({variables: {assetKeys: nextChunk.map(asAssetKeyInput)}});
+      const data = result.data?.wipeAssets;
+      switch (data?.__typename) {
+        case 'AssetNotFoundError':
+          showCustomAlert({
+            title: 'Could not wipe asset materializations',
+            body: 'Asset not found.',
+          });
+          break;
+        case 'AssetWipeSuccess':
+          setWiped((wiped) => wiped + nextChunk.length);
+          break;
+        case 'PythonError':
+          showCustomAlert({
+            title: 'Could not wipe asset materializations',
+            body: <PythonErrorInfo error={data} />,
+          });
+          break;
+        case 'UnauthorizedError':
+          showCustomAlert({
+            title: 'Could not wipe asset materializations',
+            body: 'You do not have permission to do this.',
+          });
+          break;
+      }
```

Comment:
It seems like there could be some risk of spewing alert dialogs, e.g. every one of the selected chunks lead to an error. Perhaps the errors could be consolidated in some way, as they are in the bulk termination dialog?